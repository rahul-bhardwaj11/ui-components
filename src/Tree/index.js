import React, { Component } from 'react';
import SortableTree from 'react-sortable-tree';
import PropTypes from 'prop-types';
import 'react-sortable-tree/style.css';
import TreeNode from './TreeNode';

class Tree extends Component {
  constructor(props) {
    super(props);
  }
  static propTypes = {
    treeData: PropTypes.array,
    onChange: PropTypes.func,
    onMoveNode: PropTypes.func,
    canDrag: PropTypes.func,
    canDrop: PropTypes.func,
    treeNodeRenderer: PropTypes.func
  };

  render() {
    return (
      <SortableTree
        {...this.props}
        maxDepth={2}
        scaffoldBlockPxWidth={10}
        isVirtualized={false}
      />
    );
  }
}
Tree.TreeNode = TreeNode;
export default Tree;
