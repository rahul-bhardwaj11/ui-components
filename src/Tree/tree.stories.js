import React from 'react';

import { storiesOf } from '@storybook/react';
import Tree from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs } from '@storybook/addon-knobs';

const stories = storiesOf('Tree', module);
stories.addDecorator(withKnobs);

const treeData = [
  {
    title: '0-0',
    key: '0-0',
    children: [
      {
        title: '0-0-0',
        key: '0-0-0'
      },
      {
        title: '0-0-1',
        key: '0-0-1'
      }
    ]
  },
  {
    title: '0-1',
    key: '0-1',
    children: [
      {
        title: '0-1-0',
        key: '0-1-0'
      }
    ]
  },
  { title: '0-2', key: '0-2' }
];

stories.add(
  'Tree',
  withInfo('Basic usage of the Tree')(() => <Tree treeData={treeData} />)
);
