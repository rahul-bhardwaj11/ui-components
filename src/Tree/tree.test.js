import React from 'react';
import renderer from 'react-test-renderer';
import Tree from './index';

describe('Tree', () => {
  it('matches default snapshot', () => {
    const treeData = [
      {
        title: '0-0',
        key: '0-0',
        children: [
          {
            title: '0-0-0',
            key: '0-0-0'
          },
          {
            title: '0-0-1',
            key: '0-0-1'
          }
        ]
      },
      {
        title: '0-1',
        key: '0-1',
        children: [
          {
            title: '0-1-0',
            key: '0-1-0'
          }
        ]
      },
      { title: '0-2', key: '0-2' }
    ];
    const onChange = jest.fn();
    const tree = renderer
      .create(<Tree treeData={treeData} onChange={onChange} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
