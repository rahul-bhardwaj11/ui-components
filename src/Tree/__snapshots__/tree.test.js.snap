// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tree matches default snapshot 1`] = `
<div
  className="rst__tree"
  style={Object {}}
>
  <div
    className="rst__node"
    style={
      Object {
        "height": 62,
      }
    }
  >
    <div
      className="rst__lineBlock rst__lineHalfHorizontalRight rst__lineHalfVerticalBottom"
      style={
        Object {
          "width": 10,
        }
      }
    />
    <div
      className="rst__nodeContent"
      style={
        Object {
          "left": 10,
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div>
          <button
            aria-label="Expand"
            className="rst__expandButton"
            onClick={[Function]}
            style={
              Object {
                "left": -5,
              }
            }
            type="button"
          />
        </div>
        <div
          className="rst__rowWrapper"
        >
          <div
            className="rst__row"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <div
              className="rst__moveHandle"
            />
            <div
              className="rst__rowContents"
            >
              <div
                className="rst__rowLabel"
              >
                <span
                  className="rst__rowTitle"
                >
                  0-0
                </span>
              </div>
              <div
                className="rst__rowToolbar"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="rst__node"
    style={
      Object {
        "height": 62,
      }
    }
  >
    <div
      className="rst__lineBlock rst__lineHalfHorizontalRight rst__lineFullVertical"
      style={
        Object {
          "width": 10,
        }
      }
    />
    <div
      className="rst__nodeContent"
      style={
        Object {
          "left": 10,
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div>
          <button
            aria-label="Expand"
            className="rst__expandButton"
            onClick={[Function]}
            style={
              Object {
                "left": -5,
              }
            }
            type="button"
          />
        </div>
        <div
          className="rst__rowWrapper"
        >
          <div
            className="rst__row"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <div
              className="rst__moveHandle"
            />
            <div
              className="rst__rowContents"
            >
              <div
                className="rst__rowLabel"
              >
                <span
                  className="rst__rowTitle"
                >
                  0-1
                </span>
              </div>
              <div
                className="rst__rowToolbar"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="rst__node"
    style={
      Object {
        "height": 62,
      }
    }
  >
    <div
      className="rst__lineBlock rst__lineHalfVerticalTop rst__lineHalfHorizontalRight"
      style={
        Object {
          "width": 10,
        }
      }
    />
    <div
      className="rst__nodeContent"
      style={
        Object {
          "left": 10,
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="rst__rowWrapper"
        >
          <div
            className="rst__row"
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <div
              className="rst__moveHandle"
            />
            <div
              className="rst__rowContents"
            >
              <div
                className="rst__rowLabel"
              >
                <span
                  className="rst__rowTitle"
                >
                  0-2
                </span>
              </div>
              <div
                className="rst__rowToolbar"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
